Source/WebCore/rendering/RenderGrid.cpp

```diff
@@ -738,7 +741,7 @@ void RenderGrid::placeItemsOnGrid(GridTrackSizingAlgorithm& algorithm, std::opti
 
         // Grid items should use the grid area sizes instead of the containing block (grid container)
         // sizes, we initialize the overrides here if needed to ensure it.
-        if (!child->hasOverridingContainingBlockContentLogicalWidth())
+        if (!child->hasOverridingContainingBlockContentLogicalWidth() && !needsMasonryLayout)
```

Comment:
Yes I believe so! This is supposed to address the following from the spec: "The [containing block](https://drafts.csswg.org/css-display-3/#containing-block) for a [grid item](https://drafts.csswg.org/css-grid-2/#grid-item) is formed by its [grid area](https://drafts.csswg.org/css-grid-2/#grid-area) in the [grid axis](https://drafts.csswg.org/css-grid-3/#grid-axis) and the grid container’s content-box in the [masonry axis](https://drafts.csswg.org/css-grid-3/#masonry-axis)"

Changing this check to either `gridMasonryColumns` or `gridColumnRows` should hopefully correct it to match this

Reply:
Here I make sure that the masonry axis is set to the container size. So for width I also check `!isMasonryColumns(style())).` For height we check `!isMasonryRows(style()))`.  I modified `GridMasonryLayout::insertIntoGridAndLayoutItem` so that the grid axis size is set to the item's area after we insert the item into the grid.